tests/queries/0_stateless/01921_datatype_date32.reference

```diff
@@ -136,7 +136,7 @@
 61885
 61885
 17216
-17217
+17218
```

Comment:
Roman, could you explain please?
This is toRelativeWeekNum for '2283-12-31', right ?

Reply:
No. For line 139 the test runs:

`select toRelativeWeekNum(toDate32('2300-12-31'))`

Inside it runs:
toDayOfWeek(toDate32('2300-12-31'))
2300-12-31 is truncated to 2299-12-31 by toDate32
So toDayOfWeek returns 7.

Consider the code from master:
`return toDayNum(i + 8 - toDayOfWeek(i)) / 7;`

In fact it does:
`2299-12-31 + 8 - 7 `

So 2299-12-31 + 8 is truncated to 2299-12-31
Then `2299-12-31 - 7 = 2299-12-24 `
so toRelativeWeekNum runs for 2299-12-24  which is 17217
while correct toRelativeWeekNum should run 2299-12-31 which is 17218

I verified the calculations in Excel:

![Screenshot from 2022-10-04 18-27-47](https://user-images.githubusercontent.com/18441691/193861242-c3393607-8785-46e0-b797-b323810520f2.png)

Reply:
[toRelativeDayNum.ods](https://github.com/ClickHouse/ClickHouse/files/9708342/toRelativeDayNum.ods)
Excel's file.

Reply:
To me the test does not make much sense, but the behavior is logical and expected, so everything is fine, thanks.
Excel rocks ;)

Reply:
Excel is the second best instrument (the first is ClickHouse:) )

Reply:
Interesting, thanks. I will place the Windows calculator on the third place :)